akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala

```diff
@@ -317,45 +319,66 @@ private[remote] class ArteryAeronUdpTransport(_system: ExtendedActorSystem, _pro
         settings.Advanced.Aeron.IdleCpuLevel < 5) 0 // also don't spin for small IdleCpuLevels
     else 50 * settings.Advanced.Aeron.IdleCpuLevel - 240
 
-  override protected def runInboundStreams(): Unit = {
-    runInboundControlStream()
-    runInboundOrdinaryMessagesStream()
+  override protected def runInboundStreams(): Future[Int] = {
```

Comment:
Would returning a `Future` vs `Unit` here become a problem? I see why it could resolve the race.

Reply:
It's not problematic from a binary compatibility perspective, because the function is `protected` and the class is `private[remote]`, so it's not part of the public API. Do you see any other potential problems?